src/main/java/com/uid2/shared/secure/gcpoidc/PolicyValidator.java

```diff
@@ -120,13 +131,24 @@ private Environment checkEnvOverrides(TokenPayload payload) throws AttestationEx
             }
         }
 
+        checkAttestationUrl(new HashMap<>(envOverrides));
+
         if(!envOverridesCopy.isEmpty()){
             throw new AttestationClientException("More env overrides than allowed. " + envOverridesCopy);
         }
 
         return env;
     }
 
+    private void checkAttestationUrl(HashMap<String, String> optionalEnvOverrides) throws AttestationException {
+        if (!Strings.isNullOrEmpty(optionalEnvOverrides.get(ENV_CORE_ENDPOINT))) {
+            String givenAttestationUrl = optionalEnvOverrides.get(ENV_CORE_ENDPOINT);
+            if (!UrlEquivalenceValidator.areUrlsEquivalent(givenAttestationUrl, this.attestationUrl, LOGGER)) {
+                throw new AttestationClientException("The given attestation URL is unknown. Given URL: " + givenAttestationUrl);
```

Comment:
We are not using `AttestationFailure` here which is inconsistent with the Nitro version. Should we fix that in Nitro?

Reply:
The nitro version is a bit different, but I have added the AttestationFailure to the AttestationClientException